eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Gov.hs

```diff
@@ -193,6 +193,8 @@ data ConwayGovPredFailure era
       (NonEmpty (Voter (EraCrypto era), GovActionId (EraCrypto era)))
   | -- | Predicate failure for votes by entities that are not present in the ledger state
     VotersDoNotExist (NonEmpty (Voter (EraCrypto era)))
+  | -- | Treasury withdrawals that sum up to zero are not allowed
+    EmptyTreasuryWithdrawals
```

Comment:
I agree, we should specify which procedure or action is causing the problem ! But what is the `Int` in the suggestion? Wouldn't GovActionId be better at identifying the problematic withdrawal?

Reply:
You don't mean `Int` as an index in the `Map (RewardAccount (EraCrypto era)) Coin)` of a TreasuryWithdrawal , no? 
Because  if there's several elements in the map, then all of them will be zero if this predicate happens.

Reply:
I went ahead with `GovAction` parameter, since we use this approach with other predicate failures that fail to validate the proposed action, but I'm happy to change it if you meant something else that I missed.

Reply:
I meant an index of the proposal in the transaction that could be captured from this line: https://github.com/IntersectMBO/cardano-ledger/blob/e3cd9c812930d15d14d0c39f1a88b42275868a89/eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Gov.hs#L430

That would unambiguous, because it would indicate the exact offender. With current approach if I propose two identical `TreasuryWithdrawals`, both of which sum up to zero a user will get two identical predicate failures :slightly_smiling_face: 

That being said, this is a really dumb example and if a user creates such transactions it's their own fault :smile: So, using the actual governance action in the proposal will be just fine.

Reply:
> Wouldn't GovActionId be better at identifying the problematic withdrawal?

`GovActionId` consists of transaction body hash and the index of the proposal. Considering that the predicate failure is reported on the transaction it does not make sense to echo its hash back to the user. Therefore ,it would have been enough to report just the index. Again, though, reporting back the full proposal is ok too.

Reply:
> an index of the proposal in the transaction
I see now, makes sense and it's a more precise identification of the problem. I will leave `GovAction`  only because it seems more consistent with other predicate failures (though there is already quite some inconsistency between them in general).